lib/libzpool/Makefile.am

```diff
@@ -12,11 +16,15 @@ AM_CFLAGS += $(NO_UNUSED_BUT_SET_VARIABLE)
 # Includes kernel code generate warnings for large stack frames
 AM_CFLAGS += $(FRAME_LARGER_THAN)
 
-AM_CFLAGS += -DLIB_ZPOOL_BUILD
+AM_CFLAGS += -DLIB_ZPOOL_BUILD -Wframe-larger-than=20480 -Wno-uninitialized
```

Comment:
`-Wframe-larger-than=20480` should get removed here; see https://github.com/zfsonlinux/zfs/pull/8044#discussion_r226807656

Reply:
We're keeping the pristine copy of libzstd, so this flag is still required.

Reply:
only if you keep zstd as is. half of the code of zstd is unused. if you keep the unused code out. the whole line is obsolete. see my variant of the zstd patch

Reply:
There's no reason for us to mangle the zstd dependency, and it would make it harder to upgrade it over time. Leaving it as is makes the maintenance burden very minor.

Reply:
Full ACK. Libs should be kept vanilla whenever possible

Reply:
@brainslayer I assume this should have gone here...

>BrainSlayer commented 14 minutes ago
>i would ack too. but there is a counter argument. the code you compile contains alot of unused code. its about 300 kb bigger than it should and the unused code cannot even be used due the high stack size usage. you compile basicly broken code which will crash any kernel if used. 

>>Am 28.07.2019 um 10:14 schrieb Michael Niewöhner:…   ***@***.**** commented on this pull request.   ------------------------------------------------------------------------   In lib/libzpool/Makefile.am  <#9024 (comment)>:   > @@ -12,11 +16,15 @@ AM_CFLAGS += $(NO_UNUSED_BUT_SET_VARIABLE)    # Includes kernel code generate warnings for large stack frames    AM_CFLAGS += $(FRAME_LARGER_THAN)   -AM_CFLAGS += -DLIB_ZPOOL_BUILD  +AM_CFLAGS += -DLIB_ZPOOL_BUILD -Wframe-larger-than=20480 -Wno-uninitialized   Full ACK. Libs should be kept vanilla whenever possible   —  You are receiving this because you commented.  Reply to this email directly, view it on GitHub  <#9024?email_source=notifications&email_token=AB2WNE5YMXU2NJC46J7Y5C3QBVIP5A5CNFSM4IB75NFKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB7Y2PGA#discussion_r307990857>,  or mute the thread  <https://github.com/notifications/unsubscribe-auth/AB2WNEZEQML3MMAYX5R6YJ3QBVIP5ANCNFSM4IB75NFA>.
--


~~~